### PR TITLE
refactor: solana walletconnect rpc interface

### PR DESCRIPTION
**Pull request by zoruka**
# Description

This PR is updating the Solana's WalletConnectConnector RPC requests to use serialized transactions instead of object to communicate with wallets. For further work we should make [adapter](https://github.com/WalletConnect/web3modal/blob/main/packages/solana/src/utils/wallet-standard/adapter.ts) and [walletConnectConnector](https://github.com/WalletConnect/web3modal/blob/main/packages/solana/src/connectors/walletConnectConnector.ts) to implement a same interface which is then used by the [library consumers](https://github.com/WalletConnect/web3modal/blob/main/apps/laboratory/src/components/Solana/SolanaSignTransactionTest.tsx#L23).

This PR is related to https://github.com/WalletConnect/web-examples/pull/672

## Type of change

- [ ] Chore (non-breaking change that addresses non-functional tasks, maintenance, or code quality improvements)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Associated Issues

For Linear issues: Closes APKT-xxx
For GH issues: closes #...

# Showcase (Optional)

https://github.com/user-attachments/assets/068d237f-1e9e-4137-b9f9-3f7d9e3e7149

# Checklist

- [x] Code in this PR is covered by automated tests (Unit tests, E2E tests)
- [x] My changes generate no new warnings
- [x] I have reviewed my own code
- [x] I have filled out all required sections

### CODE REVIEW

**Review by glitch-txs — APPROVED** [review]

File: apps/laboratory/src/components/Solana/SolanaSendTransactionTest.tsx [hunk 1]
```diff
@@ -6,16 +6,15 @@ import {
...
+  SystemProgram
 } from '@solana/web3.js'
 
 import { solana } from '../../utils/ChainsUtil'
 import { useChakraToast } from '../Toast'
 
 const PHANTOM_TESTNET_ADDRESS = '8vCyX7oB6Pc3pbWMGYYZF5pbSnAdQ7Gyr32JqxqCy8ZR'
 const recipientAddress = new PublicKey(PHANTOM_TESTNET_ADDRESS)
-const amountInLamports = 100000000
+const amountInLamports = 10000000
```
Comment (enesozturk) [inline apps/laboratory/src/components/Solana/SolanaSendTransactionTest.tsx:17]: Can we use underscores for better readability?

```suggestion
const amountInLamports = 10_000_000
```

File: apps/laboratory/src/components/Solana/SolanaSignAndSendTransactionTest.tsx [hunk 2]
```diff
@@ -1,22 +1,28 @@
...
+  VersionedTransaction
+} from '@solana/web3.js'
 
 import { solana } from '../../utils/ChainsUtil'
 import { useChakraToast } from '../Toast'
 
 const PHANTOM_TESTNET_ADDRESS = '8vCyX7oB6Pc3pbWMGYYZF5pbSnAdQ7Gyr32JqxqCy8ZR'
 const recipientAddress = new PublicKey(PHANTOM_TESTNET_ADDRESS)
-const amountInLamports = 50000000
+const amountInLamports = 10000000
```
Comment (enesozturk) [inline apps/laboratory/src/components/Solana/SolanaSignAndSendTransactionTest.tsx:17]: Same here

File: packages/solana/src/connectors/baseConnector.ts [hunk 3]
```diff
@@ -36,15 +35,10 @@ export interface Connector {
...
-  signMessage: (message: Uint8Array) => Promise<string>
-  signTransaction: (
-    transaction: Transaction | VersionedTransaction
-  ) => Promise<{ signatures: { signature: string }[] }>
-  sendTransaction: (transaction: Transaction | VersionedTransaction) => Promise<string>
-  signAndSendTransaction: (
-    transaction: Transaction | VersionedTransaction,
-    options?: SendOptions
-  ) => Promise<TransactionSignature>
+  signMessage: Provider['signMessage']
```
Comment (enesozturk) [inline packages/solana/src/connectors/baseConnector.ts:38]: 💯

File: packages/solana/src/connectors/baseConnector.ts [hunk 4]
```diff
@@ -36,15 +35,10 @@ export interface Connector {
...
+  signTransaction: Provider['signTransaction']
+  signAndSendTransaction: Provider['signAndSendTransaction']
+  sendTransaction: Provider['sendTransaction']
```
Comment (enesozturk) [inline packages/solana/src/connectors/baseConnector.ts:41]: 💯

**Review by enesozturk — APPROVED** [review]

File: apps/laboratory/src/components/Solana/SolanaTests.tsx [hunk 5]
```diff
@@ -49,13 +50,23 @@ export function SolanaTests() {
           </Box>
           <Box>
             <Heading size="xs" textTransform="uppercase" pb="2">
-              Sign and Send Transaction (dApp)
+              Sign and Send Transaction (Dapp)
+              <Tooltip label="The transaction will be signed by the Wallet, returned to the Dapp and the Dapp will send the transaction into the network">
+                <Text as="span" fontSize="sm" ml="2">
```
Comment (tomiir) [inline apps/laboratory/src/components/Solana/SolanaTests.tsx:55]: nice

File: packages/solana/src/connectors/walletConnectConnector.ts [hunk 6]
```diff
@@ -1,20 +1,20 @@
 import base58 from 'bs58'
-import { PublicKey, Transaction, VersionedTransaction, type SendOptions } from '@solana/web3.js'
+import { Connection, Transaction, VersionedTransaction, type SendOptions } from '@solana/web3.js'
 import { OptionsController } from '@web3modal/core'
 
 import { SolStoreUtil } from '../utils/scaffold/index.js'
 import { UniversalProviderFactory } from './universalProvider.js'
-import { BaseConnector } from './baseConnector.js'
+import { BaseConnector, type Connector } from './baseConnector.js'
```
Comment (tomiir) [inline packages/solana/src/connectors/walletConnectConnector.ts:7]: split regular imports and type imports:
```suggestion
import { BaseConnector } from './baseConnector.js'
import type { Connector } from './baseConnector.js'
```

**Pull request merged**